receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,242 @@
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.opentelemetry.io/collector/component"
+)
+
+// combindedLogsReceiver wraps alerts and log receivers in a single log receiver to be consumed by the factory
+type combindedLogsReceiver struct {
+	alerts *alertsReceiver
+	logs   *receiver
+}
+
+type resourceInfo struct {
+	Org      *mongodbatlas.Organization
+	Project  *mongodbatlas.Project
+	Cluster  *mongodbatlas.Cluster
+	Hostname string
+	LogName  string
+}
+
+const collection_interval = time.Minute * 5
+
+func (c *combindedLogsReceiver) Start(ctx context.Context, host component.Host) error {
+
+	var errs error
+
+	// If we have an alerts receiver start alerts collection
+	if c.alerts != nil {
+		if err := c.alerts.Start(ctx, host); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	// If we have a logging receiver start log collection
+	if c.logs != nil {
+		if err := c.logs.Start(ctx, host); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func (c *combindedLogsReceiver) Shutdown(ctx context.Context) error {
+	var errs error
+
+	// If we have an alerts receiver shutdown alerts collection
+	if c.alerts != nil {
+		if err := c.alerts.Shutdown(ctx); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	// If we have a logging receiver shutdown log collection
+	if c.logs != nil {
+		if err := c.logs.Shutdown(ctx); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func (s *receiver) createProjectMap() (map[string]*Project, error) {
+	// create a map of Projects
+	projects := make(map[string]*Project)
+
+	for _, project := range s.cfg.Logs.Projects {
+		projects[project.Name] = project
+	}
+
+	return projects, nil
+}
+
+// function parses out the hostname from the specified cluster host
+func FilterHostName(s string) []string {
+	var hostnames []string
+
+	// first check to make sure string is of adequate size
+	if len(s) > 0 {
+		// create an array with a comma delimiter from the original string
+		tmp := strings.Split(s, ",")
+		for _, t := range tmp {
+
+			u, err := url.Parse(t)
+			if err != nil {
+				fmt.Printf("Error parsing out %s", t)
+			}
+
+			// separate hostname from scheme and port
+			host, _, err := net.SplitHostPort(u.Host)
+			if err != nil {
+				// the scheme prefix was not added on to this string
+				// thus the hostname will have been placed under the "Scheme" field
+				hostnames = append(hostnames, u.Scheme)
+			} else {
+				// hostname parsed successfully
+				hostnames = append(hostnames, host)
+			}
+
+		}
+	}
+
+	return hostnames
+}
+
+func (s *receiver) KickoffReceiver(ctx context.Context) {
+	stopper := make(chan struct{})
+	s.stopperChanList = append(s.stopperChanList, stopper)
+	cfgProjects, err := s.createProjectMap()
+	if err != nil {
+		s.log.Debug("error ")
```

Comment:
You should Include the error in this message, and it should also be an error message
```suggestion
		s.log.Error("Failed to create project map", zap.Error(err))
```